### 데이터 불러오기

In [1]:
import numpy as np
import pandas as pd

In [2]:
train = pd.read_csv('train.csv', index_col = 'no')
test = pd.read_csv('test.csv', index_col = 'no')

train

age          workclass  fnlwgt      education  education-num  \
no                                                                    
1       25            Private  219199           11th              7   
2       39            Private   52978   Some-college             10   
3       35            Private  196899      Bachelors             13   
4       64            Private  135527      Assoc-voc             11   
5       24            Private   60783   Some-college             10   
...    ...                ...     ...            ...            ...   
29301   20            Private  100605        HS-grad              9   
29302   21            Private  372636        HS-grad              9   
29303   18   Self-emp-not-inc  258474           10th              6   
29304   33            Private  157446           11th              7   
29305   65                  ?   94809        HS-grad              9   

            marital-status          occupation    relationship  \
no                                                               
1                 Divorced   Machine-op-inspct   Not-in-family   
2                 Divorced       Other-service   Not-in-family   
3            Never-married   Handlers-cleaners   Not-in-family   
4                 Divorced        Tech-support   Not-in-family   
5       Married-civ-spouse    Transport-moving         Husband   
...                    ...                 ...             ...   
29301        Never-married               Sales       Own-child   
29302        Never-married               Sales       Own-child   
29303        Never-married     Farming-fishing       Own-child   
29304        Never-married        Craft-repair   Not-in-family   
29305              Widowed                   ?   Not-in-family   

                      race      sex  capital-gain  capital-loss  \
no                                                                
1                    White     Male             0             0   
2                    White   Female             0          1721   
3       Asian-Pac-Islander   Female             0             0   
4                    White   Female             0             0   
5                    White     Male             0             0   
...                    ...      ...           ...           ...   
29301                Other     Male             0             0   
29302                Black     Male             0             0   
29303                White     Male             0             0   
29304                White     Male             0             0   
29305                White   Female             0             0   

       hours-per-week  native-country  income  
no                                             
1                  40   United-States       0  
2                  55   United-States       0  
3                  50           Haiti       0  
4                  40   United-States       0  
5                  70   United-States       1  
...               ...             ...     ...  
29301              40     Puerto-Rico       0  
29302              40   United-States       0  
29303              40   United-States       0  
29304              65   United-States       0  
29305              40   United-States       0  

[29305 rows x 15 columns]

In [3]:
test

age          workclass  fnlwgt      education  education-num  \
no                                                                    
29306   18                  ?  245274   Some-college             10   
29307   29            Private   83003        HS-grad              9   
29308   45            Private   35136      Bachelors             13   
29309   42   Self-emp-not-inc   64631      Bachelors             13   
29310   41            Private  195821      Doctorate             16   
...    ...                ...     ...            ...            ...   
48838   45   Self-emp-not-inc  116789        HS-grad              9   
48839   48            Private  185079        HS-grad              9   
48840   63            Private  117473   Some-college             10   
48841   18            Private  150817           11th              7   
48842   31            Private  341632     Assoc-acdm             12   

            marital-status        occupation    relationship    race      sex  \
no                                                                              
29306        Never-married                 ?       Own-child   White     Male   
29307   Married-civ-spouse     Other-service            Wife   White   Female   
29308   Married-civ-spouse      Tech-support         Husband   Black     Male   
29309   Married-civ-spouse   Exec-managerial         Husband   White     Male   
29310   Married-civ-spouse   Exec-managerial            Wife   White   Female   
...                    ...               ...             ...     ...      ...   
48838   Married-civ-spouse      Craft-repair         Husband   White     Male   
48839        Never-married   Exec-managerial   Not-in-family   White   Female   
48840   Married-civ-spouse    Prof-specialty         Husband   White     Male   
48841        Never-married             Sales       Own-child   White   Female   
48842   Married-civ-spouse      Tech-support         Husband   Black     Male   

       capital-gain  capital-loss  hours-per-week  native-country  
no                                                                 
29306             0             0              16   United-States  
29307             0             0              40   United-States  
29308             0             0              40   United-States  
29309             0             0              40   United-States  
29310             0          1902              40   United-States  
...             ...           ...             ...             ...  
48838             0             0              60   United-States  
48839             0             0              50   United-States  
48840          4386             0              40   United-States  
48841             0             0              20   United-States  
48842             0             0              46   United-States  

[19537 rows x 14 columns]

In [4]:
print(train.shape)
print(test.shape)

(29305, 15)
(19537, 14)


# 데이터 전처리

### 결측치 파악하기

범주형 결측치 파악

In [5]:
train['workclass'].value_counts()                 # workclass 결측치 있음, 4번째로 많은 범주에 해당

 Private             20410
 Self-emp-not-inc     2305
 Local-gov            1868
 ?                    1663
 State-gov            1201
 Self-emp-inc          987
 Federal-gov           854
 Without-pay            12
 Never-worked            5
Name: workclass, dtype: int64

In [6]:
test['workclass'].value_counts()         

 Private             13496
 Self-emp-not-inc     1557
 Local-gov            1268
 ?                    1136
 State-gov             780
 Self-emp-inc          708
 Federal-gov           578
 Without-pay             9
 Never-worked            5
Name: workclass, dtype: int64

In [7]:
train['education'].value_counts()

 HS-grad         9435
 Some-college    6558
 Bachelors       4848
 Masters         1587
 Assoc-voc       1230
 11th            1057
 Assoc-acdm       980
 10th             816
 7th-8th          574
 Prof-school      500
 9th              430
 12th             415
 Doctorate        347
 5th-6th          334
 1st-4th          148
 Preschool         46
Name: education, dtype: int64

In [8]:
train['marital-status'].value_counts()

 Married-civ-spouse       13331
 Never-married             9673
 Divorced                  4030
 Widowed                    942
 Separated                  914
 Married-spouse-absent      394
 Married-AF-spouse           21
Name: marital-status, dtype: int64

In [9]:
train['occupation'].value_counts()                  # occupation  결측치 있음, 중간정도로 많은 범주에 해당

 Prof-specialty       3724
 Craft-repair         3632
 Exec-managerial      3609
 Adm-clerical         3440
 Sales                3292
 Other-service        2975
 Machine-op-inspct    1811
 ?                    1668
 Transport-moving     1393
 Handlers-cleaners    1249
 Farming-fishing       888
 Tech-support          865
 Protective-serv       603
 Priv-house-serv       146
 Armed-Forces           10
Name: occupation, dtype: int64

In [10]:
test['occupation'].value_counts()                  # occupation  결측치 있음, 중간정도로 많은 범주에 해당

 Craft-repair         2480
 Exec-managerial      2477
 Prof-specialty       2448
 Sales                2212
 Adm-clerical         2171
 Other-service        1948
 Machine-op-inspct    1211
 ?                    1141
 Transport-moving      962
 Handlers-cleaners     823
 Farming-fishing       602
 Tech-support          581
 Protective-serv       380
 Priv-house-serv        96
 Armed-Forces            5
Name: occupation, dtype: int64

In [11]:
train['relationship'].value_counts()

 Husband           11769
 Not-in-family      7637
 Own-child          4554
 Unmarried          3071
 Wife               1370
 Other-relative      904
Name: relationship, dtype: int64

In [12]:
train['race'].value_counts()                      

 White                 25081
 Black                  2776
 Asian-Pac-Islander      919
 Amer-Indian-Eskimo      290
 Other                   239
Name: race, dtype: int64

In [13]:
train['sex'].value_counts()

 Male      19621
 Female     9684
Name: sex, dtype: int64

In [14]:
train['native-country'].value_counts()             # native-country 결측치 있음, 3번째로 많은 범주에 해당

 United-States                 26301
 Mexico                          579
 ?                               495
 Philippines                     178
 Germany                         124
 Puerto-Rico                     117
 Canada                          111
 El-Salvador                      96
 Cuba                             94
 India                            90
 England                          75
 South                            74
 China                            68
 Jamaica                          68
 Dominican-Republic               61
 Italy                            60
 Japan                            54
 Poland                           53
 Columbia                         53
 Guatemala                        51
 Haiti                            51
 Vietnam                          50
 Portugal                         43
 Taiwan                           39
 Iran                             38
 Nicaragua                        28
 Greece                           25
 

In [15]:
test['native-country'].value_counts()             # native-country 결측치 있음, 3번째로 많은 범주에 해당

 United-States                 17531
 Mexico                          372
 ?                               362
 Philippines                     117
 Germany                          82
 Canada                           71
 Puerto-Rico                      67
 India                            61
 El-Salvador                      59
 China                            54
 England                          52
 Italy                            45
 Cuba                             44
 Dominican-Republic               42
 South                            41
 Japan                            38
 Jamaica                          38
 Guatemala                        37
 Vietnam                          36
 Poland                           34
 Columbia                         32
 Taiwan                           26
 Ecuador                          24
 Greece                           24
 Portugal                         24
 Haiti                            24
 Peru                             21
 

In [16]:
train['income'].value_counts()

0    22260
1     7045
Name: income, dtype: int64

- 숫자형 데이터 결측치 파악

In [17]:
train

age          workclass  fnlwgt      education  education-num  \
no                                                                    
1       25            Private  219199           11th              7   
2       39            Private   52978   Some-college             10   
3       35            Private  196899      Bachelors             13   
4       64            Private  135527      Assoc-voc             11   
5       24            Private   60783   Some-college             10   
...    ...                ...     ...            ...            ...   
29301   20            Private  100605        HS-grad              9   
29302   21            Private  372636        HS-grad              9   
29303   18   Self-emp-not-inc  258474           10th              6   
29304   33            Private  157446           11th              7   
29305   65                  ?   94809        HS-grad              9   

            marital-status          occupation    relationship  \
no                                                               
1                 Divorced   Machine-op-inspct   Not-in-family   
2                 Divorced       Other-service   Not-in-family   
3            Never-married   Handlers-cleaners   Not-in-family   
4                 Divorced        Tech-support   Not-in-family   
5       Married-civ-spouse    Transport-moving         Husband   
...                    ...                 ...             ...   
29301        Never-married               Sales       Own-child   
29302        Never-married               Sales       Own-child   
29303        Never-married     Farming-fishing       Own-child   
29304        Never-married        Craft-repair   Not-in-family   
29305              Widowed                   ?   Not-in-family   

                      race      sex  capital-gain  capital-loss  \
no                                                                
1                    White     Male             0             0   
2                    White   Female             0          1721   
3       Asian-Pac-Islander   Female             0             0   
4                    White   Female             0             0   
5                    White     Male             0             0   
...                    ...      ...           ...           ...   
29301                Other     Male             0             0   
29302                Black     Male             0             0   
29303                White     Male             0             0   
29304                White     Male             0             0   
29305                White   Female             0             0   

       hours-per-week  native-country  income  
no                                             
1                  40   United-States       0  
2                  55   United-States       0  
3                  50           Haiti       0  
4                  40   United-States       0  
5                  70   United-States       1  
...               ...             ...     ...  
29301              40     Puerto-Rico       0  
29302              40   United-States       0  
29303              40   United-States       0  
29304              65   United-States       0  
29305              40   United-States       0  

[29305 rows x 15 columns]

In [18]:
is_question = train['age'] == " ?"
train[is_question]                           # 모든 숫자형 컬럼 확인 결과 결측치 없음!!

Empty DataFrame
Columns: [age, workclass, fnlwgt, education, education-num, marital-status, occupation, relationship, race, sex, capital-gain, capital-loss, hours-per-week, native-country, income]
Index: []

## 결측치 처리

workclass : 1663, 
occupation : 1668, 
native-country : 495 결측치 존재

In [19]:
# 결측치 채우기2 : 그대로(Others)

train['workclass'] = train['workclass'].replace(" ?", "Others")
print(train['workclass'].value_counts())                     # train의 workclass 결측치: ?  -> Others로 변경 됐다!

test['workclass'] = test['workclass'].replace(" ?", "Others")
print(test['workclass'].value_counts())                     # test의 workclass 결측치: ?  -> Others로 변경 됐다!

train['occupation'] = train['occupation'].replace(" ?", "Others")
print(train['occupation'].value_counts())                         # occupation 결측치: ?  -> Others로 변경 됐다!

test['occupation'] = test['occupation'].replace(" ?", "Others")
print(test['occupation'].value_counts())                         # test의 occupation 결측치: ?  -> Others로 변경 됐다!

train['native-country'] = train['native-country'].replace(" ?", "Others")
print(train['native-country'].value_counts())                           # native-country 결측치: ?  -> Others로 변경 됐다!

test['native-country'] = test['native-country'].replace(" ?", "Others")
print(test['native-country'].value_counts())                           # test의 native-country 결측치: ?  -> Others로 변경 됐다!

 Private             20410
 Self-emp-not-inc     2305
 Local-gov            1868
Others                1663
 State-gov            1201
 Self-emp-inc          987
 Federal-gov           854
 Without-pay            12
 Never-worked            5
Name: workclass, dtype: int64
 Private             13496
 Self-emp-not-inc     1557
 Local-gov            1268
Others                1136
 State-gov             780
 Self-emp-inc          708
 Federal-gov           578
 Without-pay             9
 Never-worked            5
Name: workclass, dtype: int64
 Prof-specialty       3724
 Craft-repair         3632
 Exec-managerial      3609
 Adm-clerical         3440
 Sales                3292
 Other-service        2975
 Machine-op-inspct    1811
Others                1668
 Transport-moving     1393
 Handlers-cleaners    1249
 Farming-fishing       888
 Tech-support          865
 Protective-serv       603
 Priv-house-serv       146
 Armed-Forces           10
Name: occupation, dtype: int64
 Craft-repair     

In [20]:
# 결측치 채우기3 : 최빈값

# workclass의 최빈값은 Private이므로 결측값을 전부 Private로 변환
# train['workclass'] = train['workclass'].replace(" ?", " Private")
# test['workclass'] = test['workclass'].replace(" ?", " Private")

# # 최빈값 : United-States(미국) 26301 => 26796
# # 결측치 495개 => United-States로 변환
# train['native-country'] = train['native-country'].replace(" ?", " United-States")
# test['native-country'] = test['native-country'].replace(" ?", " United-States")

# 전처리 시작

## 0. fnlwgt 삭제

In [21]:
train.drop('fnlwgt', axis = 1, inplace=True)
test.drop('fnlwgt', axis = 1, inplace=True)

## 1. capital gain loss 전처리

In [22]:
train['captial'] = train['capital-gain'] - train['capital-loss']
test['captial'] = test['capital-gain'] - test['capital-loss']
train.drop('capital-gain', axis = 1, inplace=True)
train.drop('capital-loss', axis = 1, inplace=True)
test.drop('capital-gain', axis = 1, inplace=True)
test.drop('capital-loss', axis = 1, inplace=True)

## 2. martial status 전처리

In [23]:
# marital-status 그룹화
title = [' Divorced', ' Never-married', ' Married-civ-spouse', ' Separated',
       ' Married-spouse-absent', ' Widowed', ' Married-AF-spouse']
convert_title = [' Divorced', ' Never-married', ' Married', ' SepAbs',
       ' SepAbs', ' Widowed', ' Married']

title_dict = dict(zip(title, convert_title))
train['marital-status'] = train['marital-status'].map(title_dict)
test['marital-status'] = test['marital-status'].map(title_dict)
# train

In [24]:
# train['education'].unique()

## 3. edu-num/ education 전처리

In [25]:
# education  삭제
train.drop('education', axis = 1, inplace=True)
test.drop('education', axis = 1, inplace=True)


title = [7, 10, 13, 11,  5,  9, 14,  3, 15,  8,  6, 12, 16,  4,  2,  1]
convert_title = [2, 10, 13, 11,  2,  9, 14,  2, 15,  2,  2, 12, 16,  2,  2,  2]

title_dict = dict(zip(title, convert_title))

train['education-num'] = train['education-num'].map(title_dict)
test['education-num'] = test['education-num'].map(title_dict)

## 4. age 전처리

In [26]:
bins_1 = [0, 19, 29, 39, 49, 59, 69, 79, 89, 90]   # 10대~90대
labels_1 = [1, 2, 3, 4, 5, 6, 7, 8, 9]

train['age'] = pd.cut(train['age'], bins=bins_1, labels=labels_1)
test['age'] = pd.cut(test['age'], bins=bins_1, labels=labels_1)

## 5. workclass 전처리

In [27]:
# wc = [' Private', ' Self-emp-not-inc', 'Others', ' Self-emp-inc', ' Local-gov', ' State-gov',  ' Federal-gov', ' Without-pay', ' Never-worked']
# convert_wc = ['Private ', 'Self-emp-not-inc', 'Others', 'Self-emp-inc'] + ['Gov']*3 + ['Not-working']*2

# wc_dict = dict(zip(wc, convert_wc))

# train['workclass'] = train['workclass'].map(wc_dict)
# test['workclass'] = test['workclass'].map(wc_dict)

## 6. race 전처리

In [28]:
#race 삭제
train.drop('race', axis = 1, inplace=True)
test.drop('race', axis = 1, inplace=True)

## 7. relationship 전처리

In [29]:
# train['relationship'] = train['relationship'].replace(" Husband", " Married")
# train['relationship'] = train['relationship'].replace(" Wife", " Married")
# test['relationship'] = test['relationship'].replace(" Husband", " Married")
# test['relationship'] = test['relationship'].replace(" Wife", " Married")

## 8. native-country 전처리

In [30]:
# def native(country):
#     if country in [' United-States', ' Cuba']:
#         return 'US'
#     elif country in [' England', ' Germany', ' Canada', ' Italy', ' France', ' Greece', ' Philippines']:
#         return 'Western'
#     elif country in [' Mexico', ' Puerto-Rico', ' Honduras', ' Jamaica', ' Columbia', ' Laos', ' Portugal', ' Haiti',
#                      ' Dominican-Republic', ' El-Salvador', ' Guatemala', ' Peru', 
#                      ' Trinadad&Tobago', ' Outlying-US(Guam-USVI-etc)', ' Nicaragua', ' Vietnam', ' Holand-Netherlands' ]:
#         return 'Poor' # no offence
#     elif country in [' India', ' Iran', ' Cambodia', ' Taiwan', ' Japan', ' Yugoslavia', ' China', ' Hong']:
#         return 'Eastern'
#     elif country in [' South', ' Poland', ' Ireland', ' Hungary', ' Scotland', ' Thailand', ' Ecuador']:
#         return 'Poland team'
    
#     else: 
#         return country
    

# train['country'] = train['native-country'].apply(native)
# test['country'] = test['native-country'].apply(native)

# del train['native-country']
# del test['native-country']

## native-country 삭제

In [31]:
del train['native-country']
del test['native-country']

## 이상치 처리1

In [32]:
train[(train['sex'] == ' Male') & (train['relationship'] == ' Wife')] = train[(train['sex'] == ' Male') & (train['relationship'] == ' Wife')].replace(' Wife', ' Husband')
train[(train['sex'] == ' Female') & (train['relationship'] == ' Husband')] = train[(train['sex'] == ' Female') & (train['relationship'] == ' Husband')].replace(' Husband', ' Wife')
test[(test['sex'] == ' Male') & (test['relationship'] == ' Wife')] = test[(test['sex'] == ' Male') & (test['relationship'] == ' Wife')].replace(' Wife', ' Husband')
test[(test['sex'] == ' Female') & (test['relationship'] == ' Husband')] = test[(test['sex'] == ' Female') & (test['relationship'] == ' Husband')].replace(' Husband', ' Wife')

# 모델 예측

### 문제, 정답 데이터로 분리

In [33]:
y_train = train['income']
X_train = train.drop('income', axis=1)
X_test = test

In [34]:
X_train.columns

Index(['age', 'workclass', 'education-num', 'marital-status', 'occupation',
       'relationship', 'sex', 'hours-per-week', 'captial'],
      dtype='object')

In [35]:
X_test.columns

Index(['age', 'workclass', 'education-num', 'marital-status', 'occupation',
       'relationship', 'sex', 'hours-per-week', 'captial'],
      dtype='object')

In [36]:
train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 29305 entries, 1 to 29305
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype   
---  ------          --------------  -----   
 0   age             29305 non-null  category
 1   workclass       29305 non-null  object  
 2   education-num   29305 non-null  int64   
 3   marital-status  29305 non-null  object  
 4   occupation      29305 non-null  object  
 5   relationship    29305 non-null  object  
 6   sex             29305 non-null  object  
 7   hours-per-week  29305 non-null  int64   
 8   income          29305 non-null  int64   
 9   captial         29305 non-null  int64   
dtypes: category(1), int64(4), object(5)
memory usage: 2.3+ MB


### 데이터 원핫 인코딩 하기

In [37]:
X_train = pd.get_dummies(X_train)
X_test = pd.get_dummies(X_test)

In [38]:
print(X_train.shape)
print(X_test.shape)

(29305, 49)
(19537, 49)


### 컬럼 개수 맞추기

In [39]:
# set(X_train.columns) - set(X_test.columns)

In [40]:
# 'native-country_ Holand-Netherlands'의 컬럼은 생기지만 실제로 컴퓨터가 데이터는 없는 것으로 인식함!
# X_test['native-country_ Holand-Netherlands'] = 0

In [41]:
# 위치 변경을 위해 'native-country_ Holand-Netherlands' 컬럼을 tmp 변수에 넣어줌

# tmp = X_train['native-country_ Holand-Netherlands']

In [42]:
# X_train.drop('native-country_ Holand-Netherlands', axis = 1, inplace=True)

In [43]:
# X_train = pd.concat([X_train, tmp],axis=1)
# X_train

### Scaler 만들기

In [44]:
from sklearn.preprocessing import StandardScaler

In [45]:
scaler = StandardScaler()
scaler.fit(X_train)

StandardScaler()

In [46]:
transform_X_train = scaler.transform(X_train)
transform_X_train

array([[-2.20709175, -0.03568546, -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315],
       [ 0.11209121,  1.1806002 , -0.36155055, ..., -0.22128561,
         1.42342038, -1.42342038],
       [ 0.98178481,  0.77517165, -0.13266033, ..., -0.22128561,
         1.42342038, -1.42342038],
       ...,
       [-2.20709175, -0.03568546, -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315],
       [-2.20709175,  1.99145731, -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315],
       [-0.17780666, -0.03568546, -0.13266033, ..., -0.22128561,
         1.42342038, -1.42342038]])

In [47]:
transform_X_test = scaler.transform(X_test)
transform_X_test

array([[ 0.11209121, -1.98174252, -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315],
       [-0.17780666, -0.03568546, -0.13266033, ...,  4.51904676,
         1.42342038, -1.42342038],
       [ 0.98178481, -0.03568546, -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315],
       ...,
       [ 0.11209121, -0.03568546,  0.45067059, ..., -0.22128561,
        -0.70253315,  0.70253315],
       [-2.20709175, -1.65739968, -0.13266033, ..., -0.22128561,
         1.42342038, -1.42342038],
       [ 0.69188694,  0.4508288 , -0.13266033, ..., -0.22128561,
        -0.70253315,  0.70253315]])

### 모델 생성하기

In [48]:
!pip install lightgbm

!pip install xgboost

In [49]:
from sklearn.model_selection import cross_val_score, train_test_split, KFold
from sklearn.metrics import classification_report
from sklearn.metrics import accuracy_score
from lightgbm import LGBMClassifier
from sklearn.ensemble import RandomForestClassifier

In [50]:
lgb_model = LGBMClassifier(n_estimators = 1000, learning_rate=0.01, random_state=123)
lgb_model.fit(transform_X_train, y_train)

LGBMClassifier(learning_rate=0.01, n_estimators=1000, random_state=123)

## 교차 검증하기

Standard Scaler를 적용한 LightGBM 모델 kfold 교차 검증

정확도와 f1 구하기

In [51]:
kfold = KFold(n_splits=10, random_state=123, shuffle=True)

In [52]:
result1 = cross_val_score(lgb_model,
                         transform_X_train,
                         y_train,
                         cv = kfold
                        )

print(result1.mean())

0.871762668799918


In [53]:
f1_result1 = cross_val_score(lgb_model, # lgb 모델
                         transform_X_train,
                         y_train,
                         cv = kfold,
                         scoring='f1'
                        )

print(f1_result1.mean())

0.7094143639320818


# Kaggle 제출용 파일 만들기

In [54]:
result = pd.read_csv('sample_submission.csv')
result

no  income
0      29306       0
1      29307       0
2      29308       0
3      29309       0
4      29310       0
...      ...     ...
19532  48838       1
19533  48839       0
19534  48840       1
19535  48841       0
19536  48842       0

[19537 rows x 2 columns]

In [56]:
pre = lgb_model.predict(transform_X_test)  # 모델로 예측

In [57]:
result['income'] = pre

In [58]:
result.to_csv('submission.csv', index=False)